src/core/main.c

```diff
@@ -2393,6 +2393,7 @@ int main(int argc, char *argv[]) {
 
         /* Save the original command line */
         save_argc_argv(argc, argv);
+        save_env();
```

Comment:
save_argc_argv just copies the int and the pointer. `saved_argv` still points at the original array. Thus, no memory is allocated and no memory can be leaked.

In save_env(), memory is allocated, and it should be freed. Please add a STATIC_DESTRUCTOR_REGISTER for `saved_env`.

Reply:
done !